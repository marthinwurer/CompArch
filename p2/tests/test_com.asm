;
; tests COM
;
; uses instructions MOV, BPT, HALT
;
; uses addressing modes 0, 6
;
	.globl	start
;
v1:	.word	1
v2:	.word	077777
v3:	.word	177777
v4:	.word	125252
v5:	.word	052525
;
. = . + 100
;
start:	mov	v1,r1
	mov	v2,r2
	mov	v3,r3
	mov	v4,r4
	mov	v5,r5
	bpt
;
	com	r1
	com	r2
	com	r3
	com	r4
	com	r5
	bpt
;
	com	r1
	com	r2
	com	r3
	com	r4
	com	r5
	bpt
;
	halt
;
	.end	start
