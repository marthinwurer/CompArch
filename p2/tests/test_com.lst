      1                                 ;
      2                                 ; tests COM
      3                                 ;
      4                                 ; uses instructions MOV, BPT, HALT
      5                                 ;
      6                                 ; uses addressing modes 0, 6
      7                                 ;
      8 000000                                   .globl  start
      9                                 ;
     10 000000  000001                  v1:      .word   1
     11 000002  077777                  v2:      .word   077777
     12 000004  177777                  v3:      .word   177777
     13 000006  125252                  v4:      .word   125252
     14 000010  052525                  v5:      .word   052525
     15                                 ;
     16 000112                          . = . + 100
     17                                 ;
     18 000112  016701  177662          start:   mov     v1,r1
     19 000116  016702  177660                   mov     v2,r2
     20 000122  016703  177656                   mov     v3,r3
     21 000126  016704  177654                   mov     v4,r4
     22 000132  016705  177652                   mov     v5,r5
     23 000136  000003                           bpt
     24                                 ;
     25 000140  005101                           com     r1
     26 000142  005102                           com     r2
     27 000144  005103                           com     r3
     28 000146  005104                           com     r4
     29 000150  005105                           com     r5
     30 000152  000003                           bpt
     31                                 ;
     32 000154  005101                           com     r1
     33 000156  005102                           com     r2
     34 000160  005103                           com     r3
     35 000162  005104                           com     r4
     36 000164  005105                           com     r5
     37 000166  000003                           bpt
     38                                 ;
     39 000170  000000                           halt
     40                                 ;
     41 000172  000112'                          .end    start


pc     = 000007 %  
r0     = 000000 %  
r1     = 000001 %  
r2     = 000002 %  
r3     = 000003 %  
r4     = 000004 %  
r5     = 000005 %  
r6     = 000006 %  
r7     = 000007 %  
sp     = 000006 %  
start    000112 RG 
v1       000000 R  
v2       000002 R  
v3       000004 R  
v4       000006 R  
v5       000010 R  
