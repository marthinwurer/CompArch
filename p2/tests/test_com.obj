000 2 01 00
002 2 ff 7f
004 2 ff ff
006 2 aa aa
008 2 55 55
04a 4 c1 1d b2 ff
04e 4 c2 1d b0 ff
052 4 c3 1d ae ff
056 4 c4 1d ac ff
05a 4 c5 1d aa ff
05e 2 03 00
060 2 41 0a
062 2 42 0a
064 2 43 0a
066 2 44 0a
068 2 45 0a
06a 2 03 00
06c 2 41 0a
06e 2 42 0a
070 2 43 0a
072 2 44 0a
074 2 45 0a
076 2 03 00
078 2 00 00
04a
