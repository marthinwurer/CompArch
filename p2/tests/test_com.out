CPU "ARCH" Simulator, 2.5a(Jan 22 2015)
-----------------------------------------

Memory sets starting address to 112
000112:  PS[00]  016701 177662  MOV 177662(R7),R1
000116:  PS[00]  016702 177660  MOV 177660(R7),R2
000122:  PS[00]  016703 177656  MOV 177656(R7),R3
000126:  PS[10]  016704 177654  MOV 177654(R7),R4
000132:  PS[10]  016705 177652  MOV 177652(R7),R5
000136:  PS[00]  000003  BPT
  R0[000000]  R1[000001]  R2[077777]  R3[177777]
  R4[125252]  R5[052525]  R6[000000]  R7[000140]
000140:  PS[00]  005101  COM R1
000142:  PS[11]  005102  COM R2
000144:  PS[11]  005103  COM R3
000146:  PS[05]  005104  COM R4
000150:  PS[01]  005105  COM R5
000152:  PS[11]  000003  BPT
  R0[000000]  R1[177776]  R2[100000]  R3[000000]
  R4[052525]  R5[125252]  R6[000000]  R7[000154]
000154:  PS[11]  005101  COM R1
000156:  PS[01]  005102  COM R2
000160:  PS[01]  005103  COM R3
000162:  PS[11]  005104  COM R4
000164:  PS[11]  005105  COM R5
000166:  PS[01]  000003  BPT
  R0[000000]  R1[000001]  R2[077777]  R3[177777]
  R4[125252]  R5[052525]  R6[000000]  R7[000170]
000170:  PS[01]  000000  HALT

Machine Halted - HALT instruction

Simulated time 117 cycles

LAST CPUObject DESTROYED; END OF SIMULATION
